common/oauthTokenManager.go

```diff
@@ -713,16 +722,135 @@ func (credInfo *OAuthTokenInfo) GetNewTokenFromMSI(ctx context.Context) (*adal.T
 	if credInfo.IdentityInfo.MSIResID != "" {
 		params.Set("msi_res_id", credInfo.IdentityInfo.MSIResID)
 	}
+
 	req.URL.RawQuery = params.Encode()
 	req.Header.Set("Metadata", "true")
+
 	// Set context.
 	req.WithContext(ctx)
-
+	// In case of some other process (Http Server) listening at 127.0.0.1:40342 , we do not want to wait forever for it to serve request
+	msiTokenHTTPClient.Timeout = 10 * time.Second
 	// Send request
 	resp, err := msiTokenHTTPClient.Do(req)
+	// Unset the timeout back
+	msiTokenHTTPClient.Timeout = 0
+	return req, resp, err
+}
+
+// isValidArcResponse checks if the key "Www-Authenticate" is unavailable in the header of an http response
+func isValidArcResponse(resp *http.Response) bool {
+	// Parameter for validity is whether "Www-Authenticaite" exists in the response header
+	// "Www-Authenticate" contains the path to the challenge token file for Arc VMs
+	wwwAuthenticateExists := false
+	if resp != nil {
```

Comment:
Minor: the two if statements can be combined into one line

Reply:
+1

Reply:
There is a possibility of resp being null, in which case if I try to access resp.header in the same line - It will throw a null pointer reference error.

Reply:
No, it won't. Conditions are checked from left to right. i.e. `if A && B { // do this }` If A is false, B won't be evaluated.

Reply:
Understood, and noted.